core/verify_and_parse_code_from_callback_uri.go

```diff
@@ -0,0 +1,43 @@
+package core
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+func VerifyAndParseCodeFromCallbackUri(callbackUri, redirectUri, state string) (string, error) {
```

Comment:
Do callbackUri and redirectUri need a type assertion?  Will strings.HasPrefix throw if callbackUri or redirectUri is not a string?

Reply:
in Golang:
```go
func VerifyAndParseCodeFromCallbackUri(callbackUri, redirectUri, state string) 
```
is equal to 
```go
func VerifyAndParseCodeFromCallbackUri(callbackUri string, redirectUri string, state string) 
```